crates/autopilot/src/infra/persistence/dto/fee_policy.rs

```diff
@@ -48,3 +48,42 @@ pub fn from_domain(
         },
     }
 }
+
+pub fn try_into_domain(
+    policy: FeePolicy,
+    quote: Option<&domain::quote::Quote>,
+) -> Result<domain::fee::Policy, Error> {
+    let policy = match policy.kind {
+        FeePolicyKind::Surplus => domain::fee::Policy::Surplus {
+            factor: policy.surplus_factor.unwrap().try_into()?,
+            max_volume_factor: policy.surplus_max_volume_factor.unwrap().try_into()?,
+        },
+        FeePolicyKind::Volume => domain::fee::Policy::Volume {
+            factor: policy.volume_factor.unwrap().try_into()?,
+        },
+        FeePolicyKind::PriceImprovement => domain::fee::Policy::PriceImprovement {
+            factor: policy.price_improvement_factor.unwrap().try_into()?,
+            max_volume_factor: policy
+                .price_improvement_max_volume_factor
+                .unwrap()
+                .try_into()?,
+            quote: {
+                let quote = quote.ok_or(Error::MissingQuote)?;
+                domain::fee::Quote {
+                    sell_amount: quote.sell_amount,
+                    buy_amount: quote.buy_amount,
+                    fee: quote.fee,
+                }
+            },
+        },
+    };
+    Ok(policy)
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("failed to convert database data to domain data {0}")]
+    Conversion(#[from] anyhow::Error),
```

Comment:
nit: maybe call `Inconsistency` to cover both data not being there + date not being parseable.

Reply:
Renamed and refactored